Extensions/PlatformBehavior/platformerobjectruntimebehavior.ts

```diff
@@ -1679,6 +1905,205 @@ namespace gdjs {
+     *
+     * It allows to check for encompassing platforms.
+     *
+     * (local to one hitbox check)
+     */
+    foundOverHead: boolean = false;
+    /**
+     * True if any edge has been found under the character top.
+     *
+     * It allows to check for encompassing platforms.
+     *
+     * (local to one hitbox check)
+     */
+    foundUnderHead: boolean = false;
+
+    initializeBeforeSearch(
+      behavior: PlatformerObjectRuntimeBehavior,
+      upwardDeltaY: float,
+      downwardDeltaY: float
+    ) {
+      let ownerMinX = Number.MAX_VALUE;
+      let ownerMaxX = -Number.MAX_VALUE;
+      let ownerMinY = Number.MAX_VALUE;
+      let ownerMaxY = -Number.MAX_VALUE;
+      for (const hitBox of behavior.owner.getHitBoxes()) {
+        for (const vertex of hitBox.vertices) {
+          ownerMinX = Math.min(ownerMinX, vertex[0]);
+          ownerMaxX = Math.max(ownerMaxX, vertex[0]);
+          ownerMinY = Math.min(ownerMinY, vertex[1]);
+          ownerMaxY = Math.max(ownerMaxY, vertex[1]);
+        }
+      }
+
+      this.ownerMinX = ownerMinX;
+      this.ownerMaxX = ownerMaxX;
+      this.headMinY = ownerMinY + upwardDeltaY;
+      this.ownerMinY = ownerMinY;
+      this.headMaxY = ownerMinY + downwardDeltaY;
+      this.floorMinY = ownerMaxY + upwardDeltaY;
+      this.ownerMaxY = ownerMaxY;
+      this.floorMaxY = ownerMaxY + downwardDeltaY;
+
+      this.allowedMinDeltaY = upwardDeltaY;
+      // Number.MAX_VALUE and not downwardDeltaY
+      // because it would means that a platform was found.
+      // see isCollidingAnyPlatform()
+      this.allowedMaxDeltaY = Number.MAX_VALUE;
+    }
+
+    initializeBeforeHitboxCheck() {
+      this.foundOverHead = false;
+      this.foundUnderHead = false;
+    }
+
+    /**
+     * Revert the search variables to a given state.
+     *
+     * This is used to revert side effect of jumpthru check.
+     * @param previousAllowedMinDeltaY
+     * @param previousAllowedMaxDeltaY
+     */
+    revertTo(previousAllowedMinDeltaY: float, previousAllowedMaxDeltaY: float) {
+      // Other members are either constants or local to an hitbox search.
+      this.allowedMinDeltaY = previousAllowedMinDeltaY;
+      this.allowedMaxDeltaY = previousAllowedMaxDeltaY;
+    }
+
+    setFloorIsTooHigh() {
+      this.allowedMinDeltaY = Number.MAX_VALUE;
+      this.allowedMaxDeltaY = -Number.MAX_VALUE;
+    }
+
+    floorIsTooHigh(): boolean {
+      // Return true when the 2 constraints are incompatible.
+      return this.allowedMinDeltaY > this.allowedMaxDeltaY;
+    }
+
+    isCollidingAnyPlatform(): boolean {
+      return this.ownerMaxY + this.allowedMaxDeltaY <= this.floorMaxY;
+    }
+
+    getFloorDeltaY(): float {
+      return this.allowedMaxDeltaY;
+    }
+
+    /**
+     * Check if the character can follow a given Y or move not to touch it
+     * and update the context with this new constraint.
+     * @param y
+     */
+    addPointConstraint(y: float): void {
```

Comment:
This is way clearer with this function and the comments + illustrations, thanks!